conan/tools/files/patches.py

```diff
@@ -110,8 +110,9 @@ def apply_conandata_patches(conanfile):
     for it in entries:
         if "patch_file" in it:
             # The patch files are located in the root src
-            patch_file = os.path.join(conanfile.folders.base_source, it.pop("patch_file"))
-            patch(conanfile, patch_file=patch_file, **it)
+            patch_file = os.path.join(conanfile.folders.base_source, it.get("patch_file"))
+            patch(conanfile, patch_file=patch_file, patch_type=it.get("patch_type"), 
```

Comment:
Why explicitly doing the get()? The ``patch()`` function is not destructive inside, but does:
```python
patch_type = kwargs.get('patch_type')
    patch_description = kwargs.get('patch_description')
```
And if it were destructive, maybe it would be possible to clone the ``it``, so it doesn't matter if it is destructive? Because if we allow ``patch()`` to be destructive for other arguments besides ``patch_type`` and description, it will fail again in the future.

Reply:
I think the issue is that `patch_file` is both a named argument in in the `patch` function, and the argument as it was passed via `kwargs` also contained a `patch_file` member, hence the `pop`, otherwise we would have got:
```
 got multiple values for keyword argument 'patch_file'
```

However I broke some tests in the process (yay for tests) - so I'll go for the copying option so that we can pass it via `kwargs` with the `patch_file` member erased, respecting other members. This way if in the future we extend what we read from as kwargs, we don't have to explicitly list it.